app/src/main/java/com/github/studydistractor/sdp/data/Event.kt

```diff
@@ -46,39 +50,42 @@ data class FirebaseEvent(
     val long: Double? = null,
     val start: String? = null,
     val end: String? = null,
-    val lateParticipation: Boolean = false,
+    val lateParticipation: Boolean? = null,
+    val numberOfPoints: Int? = null,
     val chatId: String? = null) {
     fun toEvent(): Event {
         return Event(
-            eventId, name!!, description!!, lat!!, long!!, start!!, end!!, lateParticipation, chatId!!
+            eventId!!,
+            name!!,
+            description!!,
+            lat!!,
+            long!!,
+            start!!,
+            end!!,
+            lateParticipation!!,
+            numberOfPoints!!,
+            chatId!!
         )
     }
 }
 
 /**
  * represent participants of a particular events
  *
- * @property eventId id of the event
  * @property participants list of userId that participate in this event
  */
 data class EventParticipants(
-    val eventId: String? = null,
-    val participants: List<String>? = null
+    val eventId: String = "",
+    val participants: Map<String, Boolean> = mapOf()

```

Comment:
Why changing the List into a Map ? What is the meaning of the boolean associated to the userId ?

Reply:
Because unfortunately there is no way to implement a list into firebase realtime :( If you found another way i'll be glad to go back to list.

Reply:
As discussed in the group, we'll be switching to the map, so I'll mark this as resolved